app/models/admin_post.rb

```diff
@@ -38,7 +38,8 @@ class AdminPost < ApplicationRecord
 
   scope :for_homepage, -> { order("created_at DESC").limit(ArchiveConfig.NUMBER_OF_ITEMS_VISIBLE_ON_HOMEPAGE) }
 
-  after_save :expire_cached_home_admin_posts
+  before_validation :inherit_translated_post_comment_permissions
```

Comment:
With this callback, calling `.valid?` on a translation post could change its comment permissions, which feels kinda unexpected. Is there any reason we use `before_validation` instead of `before_save`?

Reply:
There's not any reason we need to, no -- I'd just recently read a suggestion that said anytime you change a value with a callback, you should use `before_validation` rather than `before_save` to make sure you weren't saving something with an invalid value. But we don't validate `comment_permissions` afaik (which is probably an issue of its own), so it doesn't really matter here.